analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -115,4 +123,21 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
         Options.SonarLintXml() is { } sonarLintXml
         && AnalysisContext.TryGetValue(sonarLintXml.GetText(), ShouldAnalyzeGeneratedProvider(Compilation.Language), out var shouldAnalyzeGenerated)
         && shouldAnalyzeGenerated;
+
+    /// <summary>
+    /// Check if the current file path is included or excuded and caches the result.
+    /// </summary>
+    private bool ShouldAnalyzeFile(Compilation compilation, XElement[] sonarLintSettings, string filePath) =>
+        fileCache.GetValue(compilation, x => new()) is { } cache
```

Comment:
Null check is not needed because you will either have the Dictionary retrieved because it already exists or the create value callback will create a new one.
```suggestion
        fileCache.GetValue(compilation, x => new()) is var cache
```